### DISCUSSION TITLE
Packaging develop branch for arch & derivatives.

### DISCUSSION THREAD

**Discussion by ahmubashshir** [OpenBangla/OpenBangla-Keyboard#225] [Ideas]
I'm planning to package obk this way,

* openbangla-keyboard-git
  * depends: qt5-base openbangla-im-git
* ibus-openbangla-git
  * depends: openbangla-keyboard-git
  * provides: openbangla-im-git
* fcitx5-openbangla-git
  * depends: openbangla-keyboard-git
  * provides: openbangla-im-git

is this ok?
@mominul @Adyel @bdeshi

**Comment by mominul** [comment]
As we will be having two main packages `ibus-openbangla` and `fcitx5-openbangla` in Ubuntu/Debian repositories, so I think having `ibus-openbangla-git` and `fcitx5-openbangla-git` would be the most preferable as the user-facing packages. So I don't think we'll need `openbangla-keyboard-git` anymore.

But it seems you have proposed to have 4 packages, can you elaborate? You had proposed to me that having 3 packages would be enough in our initial discussion.

**Reply by ahmubashshir** [reply]
the 4th one is virtual, it provides a common name to share between packages.
if user wants to openbangla-im, pacman will ask which variant of obk they want to use, fcitx5 or ibus?
And as openbangla-keyboard-git depends on openbangla-im, pacman will ask the same question if they install openbangla-keyboard-git.

OTOH, if they directly install the variants ({ibus,fcitx5}-openbangla-git) pacman will pull openbangla-keyboard-git automatically w/o asking anything. the previous answer has an asciinema record showing how pacman resolves this.

**Comment by ahmubashshir** [comment]
What about the common files then, like keyboard layout files and 
dictionaries?
Cause if both ibus-openbangla and fcitx5-openbangla provides them, 
they'll conflict.
A friend of mine uses both kde and xfce in same box, fcitx5 in kde, 
ibus in xfce.

openbangla-keyboard-git provides common files:
/usr/share/openbangla-keyboard/data/
/usr/share/openbangla-keyboard/icons/
/usr/share/openbangla-keyboard/layouts/


ibus-openbangla-git will provide only directly ibus dependent files
fcitx5-openbangla-git will provide only directly fcitx5 dependent files

In ubuntu/debian term,
openbangla-keyboard-git is equivalent toopenbangla-common


In a nutshell. I'm proposing to provide 3 packages.

openbangla-keyboard-git:
	provides common files and depends on a virtual(ie. nonexistent) 
package named `openbangla-im` (it can also be named 
openbangla-input-method).
	As openbangla-im is nonexistent, pacman will search which packages 
provides `openbangla-im` and ask user to choose package they want to 
install

ibus-openbangla:
	provides ibus input method, this package provides openbangla-im. also 
depends on openbangla-keyboard-git for common files.

fcitx5-openbangla:
	provides fcitx5 input method, this package provides openbangla-im. 
also depends on openbangla-keyboard-git for common files.

packaging in action: <https://asciinema.org/a/423148>
	Mubashshir

**Reply by ahmubashshir** [reply]
btw... all 3 packages are built into my local [repo](https://lazy-rat-80.telebit.io/repo/)
